### ISSUE TITLE
`testOnly -- include tag` console output missing line breaks

### ISSUE THREAD

**Issue by ericluria** [etorreborre/specs2#491]
Using `testOnly -- include tag` yields the following console output:
```
[my-lib] $ testOnly specs.EquationSpec -- include fn1
[info] EquationSpec+ return a product with its converted base price
```

I expect the output to look like it does when using `-- exclude`:
```
[my-lib] $ testOnly specs.EquationSpec -- exclude fn1
[info] EquationSpec
[info] 
[info] The pricing equations should
[info]   My Spec
[info] 
[info]     + return a product with its converted base price, tax included
[info]     + return a product with its converted base price, tax and duties included
```

If this behavior is a bug, I'd be happy to submit a PR that fixes it if you could point me in the right direction.

Thank you!

**Comment by etorreborre** [comment]
specs2 is a "stream-based" specification, meaning that it is really a stream of "Fragments" which can be either things to display (`Text`), things to execute (`Example`) or things to do some display (`Br`). In this setting, a `Tag` is just another kind of `Fragment` which applies to other fragments, before or after it, depending on the kind of specification, if it is an immutable or a mutable one.

This makes the selection of fragments to execute based on tag names a bit involved. Most of the logic can be found in `org.specs2.specification.process.DefaultSelector`. 

Possibly the easiest way to fix this issue is to:

 - exclude `Br` fragments for the filtering in [line 70](https://github.com/etorreborre/specs2/blob/master/core/src/main/scala/org/specs2/specification/process/DefaultSelector.scala#L70)

 - and probably add a bit of post-processing on [line 74](https://github.com/etorreborre/specs2/blob/master/core/src/main/scala/org/specs2/specification/process/DefaultSelector.scala#L74) to remove too many consecutive `Br` fragments

`Br` fragments are the fragments which description is a [`Br` object](https://github.com/etorreborre/specs2/blob/master/core/src/main/scala/org/specs2/specification/core/Description.scala#L77)

Note also that the `DefaultSelector` class is using `scalaz-stream`, this is why you see code like `Process1[Fragment, Fragment]` (actually the full spec is a `Process[Task, Fragment]`.

This has been annoying me for a while too so if you fix this issue I will be very happy :-).

BTW the specification containing tests for the tagging behaviour is `org.specs2.specification.SelectorSpec` (in the `core`) module. It is an unusual specification using "Group" which is something I was experimenting with at the time. Basically the idea of not having to put references to some code in the text of an immutable specification but just having a way to parse some text and "map" it to some code later on. Each section in the text corresponds to a "new group" and there should be one `eg` per `+ example` in the text. As you will see this is actually a bit annoying because you have to count the number of examples in a group to know which one to modify (not all experimentations succeed :-)).

Please ask if you have questions.

**Comment by ericluria** [comment]
Wow, Eric, thank you for the detailed response! The fix seems more complicated than I had expected, but I'll take a look and see if I can make any progress.

**Comment by etorreborre** [comment]
@ericluria did you get a chance to try to fix this?

**Comment by ericluria** [comment]
@etorreborre Unfortunately, no. I've been quite busy with work lately and haven't had a chance.

**Comment by etorreborre** [comment]
@ericluria I had a quick go at this in `3.8.4-20160723115437-39456c4`. For now this leaves some unnecessary white text in the spec and it should be packed later on. But this is better than the previous situation.

**Comment by etorreborre** [comment]
fixed with afcd80d.

**Comment by ericluria** [comment]
Awesome! :+1:
